### ISSUE TITLE
`this.unsafeWindow` no longer points to `unsafeWindow`

### ISSUE THREAD

**Issue by SukkaW** [Tampermonkey/tampermonkey#1464]
(Please fill out the issue template with your details)

## Expected Behavior

`this.unsafeWindow === unsafeWindow` and `globalThis.unsafeWindow === unsafeWindow` should be `true`.

## Actual Behavior

Got `false` instead.

## Specifications

- Chromium: 99.0.4844.84 (Official Build) (x86_64) 
- TM: 4.16.6158 Beta
- OS: macOS 12.3

## Script

(Please give an example of the script if applicable.)

```js
// ==UserScript==
// @name         New Userscript
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://example.com/
// @icon         https://icons.duckduckgo.com/ip2/example.com.ico
// @grant        unsafeWindow
// ==/UserScript==

console.log(this.unsafeWindow === unsafeWindow);
console.log(globalThis.unsafeWindow === unsafeWindow);
console.log(this === globalThis);
```

**Comment by derjanb** [comment]
Good catch! Please set the option "Add GM functions to this or window" at the script's settings tab to "On". "Auto" will detect this at the next BETA release.

**Comment by derjanb** [comment]
Should be fixed at 4.16.6159 (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|[xpi](https://firefox.tampermonkey.net/firefox-current-beta.xpi))

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions`.
Firefox users please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` 

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.
